File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        
        vector<bool> result;
        int maxCandy = *std::max_element(candies.begin(), candies.end());
        int threshold = maxCandy - extraCandies;
        for (int i = 0; i < candies.size(); i++){
            if (candies[i] >= threshold)
                result.push_back(true);
            else
                result.push_back(false);
        }
        return result;

    }
};

File: array-string/1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.py
from typing import List

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        # if already max => always yes
        # if not already max => check diff between max => if diff is same as or smaller than extraCandies, yes / if not no.
        result = []
        max_candy = max(candies)
        for i in range(len(candies)):
            if (candies[i] == max_candy) or ((max_candy - candies[i]) <= extraCandies):
                result.append(True)
            else:
                result.append(False)
        return result